newm-shared/src/main/kotlin/io/newm/shared/exposed/ExposedArrays.kt

```diff
@@ -20,91 +30,43 @@ import java.sql.Array as SQLArray
+): Column<List<Int>> = array(name, IntegerColumnType(), size)
 
 fun Table.longArray(
     name: String,
     size: Int? = null
-): Column<Array<Int>> = array(name, LongColumnType(), size)
+): Column<List<Long>> = array(name, LongColumnType(), size)
 
 fun Table.floatArray(
     name: String,
     size: Int? = null
-): Column<Array<Float>> = array(name, FloatColumnType(), size)
+): Column<List<Float>> = array(name, FloatColumnType(), size)
 
 fun Table.doubleArray(
     name: String,
     size: Int? = null
-): Column<Array<Double>> = array(name, DoubleColumnType(), size)
+): Column<List<Double>> = array(name, DoubleColumnType(), size)
 
 fun Table.uuidArray(
     name: String,
     size: Int? = null
-): Column<Array<UUID>> = array(name, UUIDColumnType(), size)
+): Column<List<UUID>> = array(name, UUIDColumnType(), size)
 
 fun Table.textArray(
     name: String,
     size: Int? = null
-): Column<Array<String>> = array(name, TextColumnType(), size)
-
-fun <T> Table.array(
-    name: String,
-    elementType: IColumnType,
-    size: Int?
-) = registerColumn<Array<T>>(name, ArrayColumnType(elementType, size))
-
-/**
- * Implementation of [ColumnType] for the SQL `ARRAY` type.
- *
- * @property elementType the type of the array element
- * @property size an optional size of the array
- * @see [Github info](https://gist.github.com/DRSchlaubi/cb146ee2b4d94d1c89b17b358187c612)
- */
-class ArrayColumnType(
-    private val elementType: IColumnType,
-    private val size: Int?
-) : ColumnType() {
-    override fun sqlType(): String = "${elementType.sqlType()} ARRAY${size?.let { "[$it]" } ?: ""}"
-
-    override fun notNullValueToDB(value: Any): Any =
-        when (value) {
-            is Array<*> -> value
-            is Collection<*> -> value.toTypedArray()
-            else -> error("Got unexpected array value of type: ${value::class.qualifiedName} ($value)")
-        }
-
-    override fun valueFromDB(value: Any): Any =
-        when (value) {
-            is SQLArray -> value.array as Array<*>
-            is Collection<*> -> value.toTypedArray()
-            else -> value
-        }
-
-    override fun setParameter(
-        stmt: PreparedStatementApi,
-        index: Int,
-        value: Any?
-    ) {
-        if (value == null) {
-            stmt.setNull(index, this)
-        } else {
-            val preparedStatement = stmt as? JdbcPreparedStatementImpl ?: error("Currently only JDBC is supported")
-            val array = preparedStatement.statement.connection.createArrayOf(elementType.sqlType(), value as Array<*>)
-            stmt[index] = array
-        }
-    }
-}
+): Column<List<String>> = array(name, TextColumnType(), size)
 
 /**
  * Invokes the `ANY` function on an expression.
  */
 @JvmName("any")
-fun <T> ExpressionWithColumnType<Array<T>>.any(): ExpressionWithColumnType<T> = anyFunc()
+fun <T> ExpressionWithColumnType<List<T>>.any(): ExpressionWithColumnType<List<T>> = anyFunc()
```

Comment:
I'm unsure if any of these work beyond the `overlaps` function. I think they are all unused as of now. The modifications are mostly to get the code to compile. If you use any of them, please test to make sure they work properly.